app/controllers/profile/orders_controller.rb

```diff
@@ -31,13 +31,18 @@ def create
 
   def update
     order = Order.find(params[:id])
-    unless order.cancelled? || order.shipped?
+    role = current_user.role
+
+    if !order.cancelled? || !order.shipped?
       order.cancel
-      flash[:success] = "Your order has been cancelled."
+      flash[:success] = 'Order cancelled'
+    end
+
+    if role != 'admin'
```

Comment:
you can drop role and use the ApplicationController helper here
```suggestion
    if !current_admin?
```